src/daemon/Display.cpp

```diff
@@ -365,27 +372,33 @@ namespace SDDM {
         return false;
     }
 
-    void Display::startAuth(const QString &user, const QString &password, const Session &session) {
+    bool Display::startAuth(const QString &user, const QString &password, const Session &session) {
 
         if (m_auth->isActive()) {
             qWarning() << "Existing authentication ongoing, aborting";
-            return;
+            return false;
         }
 
         m_passPhrase = password;
 
         // sanity check
         if (!session.isValid()) {
             qCritical() << "Invalid session" << session.fileName();
-            return;
+            return false;
         }
         if (session.xdgSessionType().isEmpty()) {
             qCritical() << "Failed to find XDG session type for session" << session.fileName();
-            return;
+            return false;
         }
         if (session.exec().isEmpty()) {
             qCritical() << "Failed to find command for session" << session.fileName();
-            return;
+            return false;
+        }
+        // check if the user even exists
+        struct passwd *pw = getpwnam(qPrintable(user));
+        if (!pw) {
```

Comment:
I think with this check the cause of the failure is never displayed to the user. Can you try to log in manually with an invalid username? It'll probably just fail quickly without any indication of the error

Reply:
Hmm.. that's right. Signal 'loginFailed' will not be emitted, so sddm-theme just stuck (disabled UI, can't press buttons), waiting signal with result of login. 

I also found out that it happened with other sanity checks, like "Failed to find command for session", even without this proposed changes. So this is a different problem. I think we need handle that in `Display::Login' and emit `loginFailed` if auth was not started ([here](https://github.com/sddm/sddm/blob/d090b034d68acc882fb9f897e95b0e297a416b41/src/daemon/Display.cpp#L331C44-L331C44)).

Also, I think maybe we should move user existence check to `Display::attemptAutologin` for security reasons, because for manual login case, user existence (is login correct) can be determined by speed of login failing. But I think that check will be alright for autologin case.

Reply:
User existence check is IMO not a sanity check. This check here can just be removed now, as all autologin auth failures are handled now.

Reply:
Ok, I will remove user existence check, but what's about handling other sanity checks? If we won't handle that, in case of manual login, the UI of sddm theme will hang and no error message will be displayed (because it waits result of authorization).

We can emit `loginFailed` signal and theme will display error indication, also we can try to say more precise reason for failure by using `m_socketServer->informationMessage(m_socket, message);` (and I think it is more suitable) with message like `System error. Authorization is not possible` or something like that. But I found out that not always sddm-themes support that `informationMessage` signal, `maya` definitely do, but `plasma breeze` handles only `loginFailed` and `loginSucceeded`.

But anyway, with `loginFailed` or `informationMessage`, I think we need to handle this, because otherwise there is no error indication (that caused by 4 sanity checks) at all in any of the sddm themes.

Reply:
> But anyway, with loginFailed or informationMessage, I think we need to handle this, because otherwise there is no error indication (that caused by 4 sanity checks) at all in any of the sddm themes.

Agreed. Sanity checks aren't really supposed to fail though so fixing this isn't high priority IMO. Should be done, but doesn't necessarily have to be in this PR.

Reply:
> Agreed. Sanity checks aren't really supposed to fail though so fixing this isn't high priority IMO. Should be done, but doesn't necessarily have to be in this PR.

Sorry, I didn't understand you completely about sanity checks. How it should be done, like I described or in some other way?

Reply:
Yes, like you described. At least `loginFailed` is mandatory.

Reply:
User existence check is removed